File: 879-maximize-distance-to-closest-person/solution.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int l = 0, r = 0, res = 1, dist = 0;
        for (int i = 0; i < seats.size(); i++) {
            if (0 == seats[i]) r ++;
            else {
                r = i;
                if (seats[l] == 0) res = r - l;
                else {
                    if ((r-l-1) % 2 == 0) dist = (r-l-1) / 2;
                    else dist = (r-l) / 2;
                    
                    if (dist > res) res = dist;              
                }
                l = i;
            }
        }
        return res > r-l ? res : r-l;
    }
};

File: 879-maximize-distance-to-closest-person/solution.py
class Solution(object):
    def maxDistToClosest(self, seats):
        """
        :type seats: List[int]
        :rtype: int
        """
        l, r, res, temp = 0, 0, 1, 0
        for i, v in enumerate(seats):
            if v == 0:
                r += 1
            elif v == 1:
                r = i
                if seats[l] == 0:
                    res = r - l
                else:
                    temp = (r-l-1) / 2 if (r-l-1) % 2 == 0 else (r-l) / 2
                    
                    if temp > res:
                        res = temp
                l = i

        return res if res > r-l else r-l
